ui/de-lib/src/main/java/org/iplantc/de/apps/client/presenter/submit/SubmitAppForPublicPresenter.java

```diff
@@ -33,56 +35,70 @@
     interface SubmitAppPresenterBeanFactory extends AutoBeanFactory {
         AutoBean<AppRefLink> appRefLink();
     }
-    @Inject PublicAppCategoryProxy appGroupProxy;
+
+    private class HierarchiesCallback implements AsyncCallback<List<OntologyHierarchy>> {
```

Comment:
The logic in these callback methods are pretty much duplicated in `OntologiesPresenterImpl` and `OntologyHierarchiesPresenterImpl`.
This logic should be refactored into a common place at some point (but not necessary for this PR).